src/core/control/xojfile/LoadHandler.cpp

```diff
@@ -407,27 +408,35 @@ void LoadHandler::parseBgPixmap() {
         this->page->setBackgroundImage(img);
     } else if (!strcmp(domain, "attach")) {
         // This is the new zip file attach domain
-        auto readResult = readZipAttachment(filepath);
+        const auto readResult = readZipAttachment(filepath);  ///< Do not remove the const qualifier - see GBytes below
         if (!readResult) {
             return;
         }
-        std::string& imgData = *readResult;
+        const std::string& imgData = *readResult;  ///< Do not remove the const qualifier - see GBytes below
 
-        GBytes* attachment = g_bytes_new_take(imgData.data(), imgData.size());
-        GInputStream* inputStream = g_memory_input_stream_new_from_bytes(attachment);
+        /**
+         * To avoid an unecessary copy, the GBytes is initialized with noop free function. The data is still managed by
+         * the std::optional<std::string> instance.
+         * The GBytes instance forbids any modification of the data until it has been released, so do not remove the
+         * const qualifier on readResult or imgData
+         */
+        GBytes* data = g_bytes_new_with_free_func(
+                imgData.data(), imgData.size(), +[](gpointer) {}, nullptr);
+        xoj::util::GObjectSPtr<GInputStream> inputStream(g_memory_input_stream_new_from_bytes(data), xoj::util::adopt);
```

Comment:
I think `g_memory_input_stream_new_from_data ` is the better choice here.

Reply:
Indeed it is!